liana/method/_Method.py

```diff
@@ -267,3 +277,204 @@ def __call__(self,
 
 def _show_methods(methods):
     return concat([method.get_meta() for method in methods])
+
+
+class MetabMethod(MethodMeta):
+    """
+    liana's Method Class
+    """
+    def __init__(self, _method, **kwargs):
+        super().__init__(method_name=_method.method_name,
+                         reference=_method.reference,
+                         complex_cols=_method.complex_cols,
+                         add_cols=_method.add_cols,
+                         fun=_method.fun,
+                         magnitude=_method.magnitude,
+                         magnitude_ascending=_method.magnitude_ascending,
+                         specificity=_method.specificity,
+                         specificity_ascending=_method.specificity_ascending,
+                         permute=_method.permute,
+                         met=_method.met
+                         )
+        self._method = _method
+        
+
+
+    def __call__(self,
+                 adata: AnnData,
+                 groupby: str,
+                 resource: Optional[DataFrame] = None,
+                 resource_name: str = 'metalinksdb',
+                 metsets_name: str = 'metalinksdb',
+                 met_est_resource: Optional[DataFrame] = None,
+                 est_fun: str = 'mean_per_cell',
+                 score_fun: str = 'cellphone',
+                 expr_prop: float = 0.1,
+                 min_cells: int = 5,
+                 base: float = 2.718281828459045,
+                 supp_columns: list = None,
+                 return_all_lrs: bool = False,
+                 prop_missing_allowed: float = 0.995,
+                 use_raw: Optional[bool] = True,
+                 layer: Optional[str] = None,
+                 verbose: Optional[bool] = False,
+                 n_perms: int = 1000,
+                 seed: int = 1337,
+                 inplace: bool = True,
+                 pass_mask: bool = True,
+                 est_only: bool = False,
+                 correct_fdr: bool = False,
+                 cellular_locations: list = ['Extracellular'],
+                 tissue_locations: list = ['Adrenal Cortex', 'Brain', 'Epidermis', 'Fibroblasts', 'Kidney', 'Neuron',
```

Comment:
Anything else I can adjust but this is too messy for me to figure out where does it even start

Reply:
I cannot figure out what is happening :)